bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,109 @@
+
+
+class WikipediaCog(commands.Cog):
+    """Get info from wikipedia."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    @staticmethod
+    def formatted_wiki_urls(index: int, titles: str) -> List[str]:
+        """Making formatted wikipedia links list."""
+        titles = f'`{index}` [{titles}]({WIKIPEDIA_URL.format(title=titles.replace(" ", "_"))})'
+        return titles
+
+    async def search_wikipedia(self, search_term: str) -> Optional[List[str]]:
+        """Search wikipedia and return the first page found."""
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+        if not search_results:
+            return None
+
+        # Ignore pages with "may refer to"
+        for search_result in search_results:
+            log.info("trying to append titles")
+            if "may refer to" not in search_result["snippet"]:
+                page.append(search_result["title"])
+        log.info("Finished appending titles")
+        return page
+
+    @commands.cooldown(1, 10, commands.BucketType.user)
+    @commands.command(name="wikipedia", aliases=["wiki"])
+    async def wikipedia_search_command(self, ctx: commands.Context, *, search: str) -> None:
+        """Return list of results containing your search query from wikipedia."""
+        titles = await self.search_wikipedia(search)
+
+        def check(message: Message) -> bool:
+            return message.author.id == ctx.author.id and message.channel == ctx.channel
+
+        if not titles:
+            await ctx.send("Sorry, we could not find a wikipedia article using that search term")
+            return
+
+        async with ctx.typing():
+            titles_no_underscore = [title.replace(" ", "_") for title in titles]  # wikipedia uses "_" as spaces
+            log.info("Finished appending titles to titles_no_underscore list")
+
+            s_desc = "\n".join(self.formatted_wiki_urls(index, title)for index, title in enumerate(titles, start=1))
+            embed = Embed(colour=Color.blue(), title=f"Wikipedia results for `{search}`", description=s_desc)
+            embed.timestamp = datetime.datetime.utcnow()
+            await ctx.send(embed=embed)
+        embed = Embed(colour=Color.green(), description="Enter number to choose")
+        msg = await ctx.send(embed=embed)
+        chances = 0
+        total_chances = Wikipedia.total_chance
+        titles_len = len(titles_no_underscore)  # getting length of list
+
+        while chances <= total_chances:
+            chances += 1
+            if chances < total_chances:
+                error_msg = f'You have `{total_chances - chances}/{total_chances}` chances left'
+            else:
+                error_msg = 'Please try again by using `.wiki` command'
```

Comment:
In addendum to using `for _ in range(Wikipedia.total_retries)`, this can probably be refactored to the following:
```python
for retry_count in range(1, Wikipedia.total_retries + 1):
    if retry_count < Wikipedia.total_retries:
        retries_left = Wikipedia.total_retries - retry_count
        error_msg = f"You have `{retries_left}/{Wikipedia.total_retries}` chances left"
    else:
        error_msg = 'Please try again by using `.wiki` command'
```
Refactor as you see fit.